src/sys/timerfd.rs

```diff
@@ -270,3 +270,14 @@ impl TimerFd {
         Ok(())
     }
 }
+
+impl Drop for TimerFd {
+    fn drop(&mut self) {
+        let result = Errno::result(unsafe {
+            libc::close(self.fd)
+        });
+        if let Err(Error::Sys(Errno::EBADF)) = result {
+            panic!("close of TimerFd encountered EBADF");
```

Comment:
One problem: panicking during `drop` is usually bad, because `drop` itself gets called during panic.  A double-panic is difficult to debug.  Instead, you should only panic if the thread isn't already panicking.  In fact, you don't even need to `close` when panicking, because the process is about to abort anyway.  So I would write this as
```rust
if !std::thread::panicking() {
    //close the file descriptor, panicking on EBADF
}
```

Reply:
The file descriptor is now only closed if the thread isn't already panicking.